lib/bitcoin/SlidingWindowQuantileCalculator.ts

```diff
@@ -0,0 +1,307 @@
+import { SlidingWindowQuantileMongoStore } from './SlidingWindowQuantileMongoStore';
+
+/**
+ * Value approximator is used to normalize values to a compact range while
+ * providing the ability to denormalize the normalized values to an approximation
+ * of the original value.
+ *
+ * For example, if we use approximation 2, the number 3 and 4 would be normalized to 3,
+ * 5,6,7,8 would be normalized to 4, and so on. Denormalization would denormalize 3 to 4,
+ * 4 to 8, and so on. If we take a number n, the number denormalize(normalize(n)) would be
+ * larger than n, but at most by a factor 2.
+ *
+ * We can replace 2 in the argument above with a value closer to 1 (say 1.1), and all values
+ * will be approximated by a factor at most 1.1.
+ */
+export class ValueApproximator {
```

Comment:
Sorry, I am failing to confidently understand the class by reading the comments above without looking into the implementation. Is this class attempting to deterministically compact a number to its "order/magnitude"? Seems that's not the case, but the example you gave is not enough for me deduct the pattern (sorry for my low IQ).

Reply:
Yes - with the guarantee that we lose a configurable amount in accuracy during this step.